### ISSUE TITLE
missing properties check not working for boolean field

### ISSUE THREAD

**Issue by myphysicslab** [google/closure-compiler#861]
When a class has a boolean field, and I misspell the name of the field, a test of the form
```
if (this.myBolean)
```
does not report an error even when using compiler option `--jscomp_error=missingProperties`.  Whereas a test of the form
```
if (this.myBolean == true)
```
will correctly report the following error:
```
compile_js.sh src/sims/engine2D/ChainApp.js build/sims/engine2D/ChainApp_en.js true false
src/lab/util/MyClass.js:14: WARNING - Property myBolean never defined on myphysicslab.lab.util.MyClass
if (this.myBolean == true)
         ^
```

Here is the example class:
```javascript
goog.provide('myphysicslab.lab.util.MyClass');
/**
@constructor
*/
myphysicslab.lab.util.MyClass = function() {
/** @type {boolean} */
this.myBoolean = true;
};

/**
@return {string}
*/
myphysicslab.lab.util.MyClass.prototype.check = function() {
if (this.myBolean)
  return "yes";
else
  return "no";
};
```

**Comment by concavelenz** [comment]
Assuming this is the old type inference (which is the tied to the missingProperties diagnostic group).  This was an attempt to allow existence checks:

if (a && a.b && a.b.c) ...

I don't personally think it was a good idea

**Comment by myphysicslab** [comment]
This is happening with new type inference also.

**Comment by myphysicslab** [comment]
Just ran into this again, it happens with numbers or strings as well, not just boolean:  Assigning to non-existent fields in a class generates no errors with NTI or old compiler.  For example, adding code like this to any class method gives no errors:
```javascript
this.fooBar_ = 42;
this.baz_ = "frumble";
```

Here is the set of options being used.  Includes `missingProperties`.
```
+ java -jar ../javascript/closure-compiler/build/compiler.jar --new_type_inf
--compilation_level=SIMPLE --closure_entry_point=myphysicslab.sims.engine2D.ChainApp
--js=src --js=../javascript/closure-library --only_closure_dependencies=true
--generate_exports --jscomp_error=checkTypes --jscomp_error=missingProperties
--jscomp_error=missingProvide --jscomp_error=missingRequire
--jscomp_warning=accessControls --jscomp_warning=ambiguousFunctionDecl
--jscomp_warning=checkVars --jscomp_warning=const --jscomp_warning=constantProperty
--jscomp_warning=fileoverviewTags --jscomp_warning=globalThis
--jscomp_warning=invalidCasts --jscomp_warning=misplacedTypeAnnotation
--jscomp_warning=missingReturn --jscomp_warning=newCheckTypes
--jscomp_warning=strictModuleDepCheck --jscomp_warning=suspiciousCode
--jscomp_warning=typeInvalidation --jscomp_warning=undefinedNames
--jscomp_warning=undefinedVars --jscomp_warning=unknownDefines
--jscomp_warning=uselessCode --jscomp_warning=visibility --extra_annotation_name=alias
--warning_level=VERBOSE --define=myphysicslab.lab.util.UtilityCore.ADVANCED=false
--define=goog.DEBUG=true '--define=goog.LOCALE='\''en'\''' --externs=my_externs.js
--language_in=ECMASCRIPT5
```

**Comment by dimvar** [comment]
Assigning has different behavior than property access, because it creates a property. If you want to prevent accidental property creation by assignment, you should annotate the constructor with <code>@struct</code>

**Comment by myphysicslab** [comment]
Thanks, I didn't know about using @struct.  Using @struct seems to also fix the original problem reported above.  I guess I'll add @struct to all my classes now.  I can't think of any classes where I add properties after creation.
